src/ProjectService.Business/Commands/CreateTaskCommand.cs

```diff
@@ -0,0 +1,118 @@
+﻿using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.Broker.Responses;
+using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Kernel.Exceptions.Models;
+using LT.DigitalOffice.Kernel.Extensions;
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+using LT.DigitalOffice.ProjectService.Business.Commands.Interfaces;
+using LT.DigitalOffice.ProjectService.Data.Interfaces;
+using LT.DigitalOffice.ProjectService.Mappers.RequestsMappers.Interfaces;
+using LT.DigitalOffice.ProjectService.Models.Db;
+using LT.DigitalOffice.ProjectService.Models.Dto.Enums;
+using LT.DigitalOffice.ProjectService.Models.Dto.Requests;
+using LT.DigitalOffice.ProjectService.Models.Dto.Responses;
+using LT.DigitalOffice.ProjectService.Validation.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LT.DigitalOffice.ProjectService.Business.Commands
+{
+    public class CreateTaskCommand : ICreateTaskCommand
+    {
+        private readonly ITaskRepository _repository;
+        private readonly ICreateTaskValidator _validator;
+        private readonly IDbTaskMapper _mapperTask;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IRequestClient<IGetDepartmentRequest> _requestClient;
+        private readonly IAccessValidator _accessValidator;
+        private readonly IProjectRepository _projectRepository;
+        private readonly ILogger<CreateTaskCommand> _logger;
+
+        private IGetDepartmentResponse GetDepartment(Guid userId, List<string> errors)
+        {
+            string errorMessage = "Cannot create task. Please try again later.";
+
+            try
+            {
+                var response = _requestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(
+                    IGetDepartmentRequest.CreateObj(userId, null)).Result;
+
+                if (response.Message.IsSuccess)
+                {
+                    return response.Message.Body;
+                }
+
+                _logger.LogWarning($"Can not find department with this id '{userId}':", userId);
```

Comment:
I did not say to write 1 in 1 as I wrote, now according to the message it is not clear why the department is searched for by the user id. But the log is now correct

Reply:
fixed